meson.build

```diff
@@ -0,0 +1,582 @@
+project('libuv', 'c',
+  version : '1.41.0',
+  license : 'libuv',
+  default_options : [ 'c_std=c89' ],
+)
+
+# get compiler type
+cc = meson.get_compiler('c')
+cc_clang = cc.get_id().contains('clang')
+cc_gnu = cc.get_id().contains('gcc')
+cc_msvc = cc.get_id().contains('msvc')
+
+# get system type
+win32 = host_machine.system() == 'windows'
+linux = host_machine.system() == 'linux'
+android = host_machine.system() == 'android'
+apple = host_machine.system() == 'darwin'
+freebsd = host_machine.system() == 'freebsd'
+netbsd = host_machine.system() == 'netbsd'
+openbsd = host_machine.system() == 'openbsd'
+dragonfly = host_machine.system() == 'dragonfly'
+haiku = host_machine.system() == 'haiku'
+sunos = host_machine.system() == 'sunos'
+qnx = cc.get_define('__QNX__') != ''
+aix = cc.get_define('_AIX') != ''
+os390 = false # TODO: detect
+os400 = false # TODO: detect
+message('System: ' + host_machine.system())
+
+# get options
+libuv_build_tests = get_option('build_tests')
+libuv_build_bench = get_option('build_benchmarks')
+libuv_qemu = get_option('build_for_qemu')
+libuv_asan = get_option('b_sanitize').contains('address')
+if not libuv_build_tests
+  libuv_build_bench = false
+endif
+
+# variable declaration
+libuv_cargs = []
+libuv_libs = []
+libuv_test_src = []
+libuv_test_libs = []
+
+# option-depending config
+if libuv_qemu
+  libuv_cargs += '-D__QEMU__=1'
+endif
+
+if libuv_asan and (cc_gnu or cc_clang)
+  libuv_cargs += '-D__ASAN__=1'
+endif
+
+# compiler flags
+if cc_msvc
+  cc_msvc_cargs_all = [
+    '/w4',      # W4
+    '/wd4100',  # no_unused_parameter
+    '/wd4127',  # no conditional constant
+    '/wd4201',  # no_nonstandard
+    '/wd4206',  # no_nonstandard_empty_tu
+    '/wd4210',  # no_nonstandard_file_scope
+    '/wd4232',  # no_nonstandard_nonstatic_dlimport
+    '/wd4456',  # no_hides_local
+    '/wd4457',  # no_hides_param
+    '/wd4459',  # no_hides_global
+    '/wd4706',  # no_conditional_assignment
+    '/wd4496',  # no_unsafe
+    '/utf-8',   # utf-8
+  ]
+
+  foreach flag : cc_msvc_cargs_all
+    if cc.has_argument(flag)
+      libuv_cargs += flag
+    endif
+  endforeach
+else
+  if cc.has_argument('-Wall')
+    libuv_cargs += '-Wall'
```

Comment:
You should not add `-Wall` or `/W4` by hand. They are added by Meson automatically based on the warning level option.